tests/snappi_tests/multidut/pfc/files/lossless_response_to_throttling_pause_storms_helper.py

```diff
@@ -0,0 +1,391 @@
+# uncompyle6 version 3.9.0
+# Python bytecode version base 2.7 (62211)
+# Decompiled from: Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
+# Embedded file name: /var/johnar/sonic-mgmt/tests/snappi/multi_dut_rdma/files/rdma_helper.py
+# Compiled at: 2023-02-10 09:15:26
+from math import ceil                                                                                      # noqa: F401
+import logging                                                                                             # noqa: F401
+from tests.common.helpers.assertions import pytest_assert, pytest_require                                  # noqa: F401
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts, fanout_graph_facts                    # noqa: F401
+from tests.common.snappi_tests.snappi_helpers import get_dut_port_id                                       # noqa: F401
+from tests.common.snappi_tests.common_helpers import pfc_class_enable_vector, stop_pfcwd, disable_packet_aging, \
+    get_pfcwd_poll_interval, get_pfcwd_detect_time, get_pfcwd_restore_time, sec_to_nanosec                 # noqa: F401
+from tests.common.snappi_tests.port import select_ports                                                    # noqa: F401
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+from tests.common.snappi_tests.traffic_generation import run_traffic, verify_pause_flow, \
+     setup_base_traffic_config, verify_m2o_oversubscribtion_results                                      # noqa: F401
+
+
+logger = logging.getLogger(__name__)
+
+PAUSE_FLOW_NAME = 'Pause Storm'
+TEST_FLOW_NAME = 'Test Flow'
+TEST_FLOW_AGGR_RATE_PERCENT = 30
+BG_FLOW_NAME = 'Background Flow'
+PAUSE_FLOW_NAME = 'PFC Traffic'
+BG_FLOW_AGGR_RATE_PERCENT = 25
+PAUSE_FLOW_RATE = 15
+DATA_PKT_SIZE = 1024
+DATA_FLOW_DURATION_SEC = 20
+DATA_FLOW_DELAY_SEC = 0
+SNAPPI_POLL_DELAY_SEC = 2
+TOLERANCE_THRESHOLD = 0.05
+PAUSE_FLOW_DURATION_SEC = 10
+PAUSE_FLOW_DELAY_SEC = 5
+
```

Comment:
please update how the test is done and expected results.

Reply:
The background traffic will include two lossy traffic streams, each with randomly chosen priorities (0..2, 5..7), and each having a 25% bandwidth. The test data traffic will consist of two lossless traffic streams, with the SONiC default lossless priorities of 3 and 4, and each having a 25% bandwidth. PFC pause throttling stream: Persistent PFC pause frames from the IXIA Rx port with enough repetition and quanta chosen so as to reduce one or all lossless streams down to 90% of their configured bandwidth.Start the background and test traffic from IXIA Tx ports 1 & 2; the combined line rate will be 100% of the egress port capacity (25% line rate bandwidth per stream)
After a few seconds, from IXIA port 3, start the PFC throttling stream onto one or all lossless priorities. Ensure that: a. The IXIA RX port indicates that the targeted lossless streams got their bandwidth reduced by 90% b. The IXIA Rx port is receiving all lossy streams and any of the lossless streams not marked for PFC pause to their nominal configured transmit bandwidth